laravel/app/Http/Controllers/ScheduleDataController.php

```diff
@@ -288,178 +285,92 @@ public function getChoice() {
-            'objective'=> null,
-            'pref_anest' => null
-        );
-        $choice++;
-        // If the second choice exists, get information for second choice
-        if ($split[1] != 0){
-            $schedule_data2 = ScheduleData::where('id', $split[1])->get();
-            $attending_string = $schedule_data2[0]['lead_surgeon'];
-            $attending = substr($attending_string, strpos($attending_string, "[")+1, strpos($attending_string, "]")-(strpos($attending_string, "[")+1));
-            $attending2 = substr($attending_string, 0, strpos($attending_string, "["));
-
-            // Store second choice data
-            $data2['schedule'] = $schedule_data2[0];
-            $data2['attending'] = $attending2;
-            // Find previous milestone and educational objective
-            $option2 = Option::where('date', $date)->where('resident', $resident)->where('option', 2)->get();
-            if (sizeof($option2) > 0){
-                $milestone2 = Milestone::where('id', $option2[0]['milestones'])->get();
-                if(sizeof($milestone2) > 0){
-                    $data2['milestone'] = $milestone2[0];
-                }
-                $data2['objective'] = $option2[0]['objectives'];
-                $data2['pref_anest'] = $option2[0]['anesthesiologist_id'];
-            }
-        }
 
-        //Initiate second choice data
-        $data3 = array(
-            'schedule' => null,
-            'attending' => null,
-            'milestone' => null,
-            'objective'=> null,
-            'pref_anest' => null
-        );
-        $choice++;
-        // If the third choice exists, get information for third choice
-        if ($split[2] != 0){
-            $schedule_data3 = ScheduleData::where('id', $split[2])->get();
-            $attending_string = $schedule_data3[0]['lead_surgeon'];
-            $attending = substr($attending_string, strpos($attending_string, "[")+1, strpos($attending_string, "]")-(strpos($attending_string, "[")+1));
-            $attending3 = substr($attending_string, 0, strpos($attending_string, "["));
-
-            // Store second choice data
-            $data3['schedule'] = $schedule_data3[0];
-            $data3['attending'] = $attending3;
-            // Find previous milestone and educational objective
-            $option3 = Option::where('date', $date)->where('resident', $resident)->where('option', 3)->get();
-            if (sizeof($option3) > 0){
-                $milestone3 = Milestone::where('id', $option3[0]['milestones'])->get();
-                if(sizeof($milestone3) > 0){
-                    $data3['milestone'] = $milestone3[0];
+        for ($i = 0; $i < 3; $i++) {
+            if ($split[$i]) {
+                $data[$i] = array(
+                    'schedule' => null,
+                    'attending' => null,
+                    'milestone' => null,
+                    'objective' => null,
+                    'pref_anest' => null
+                );
```

Comment:
Same questions as prior regarding creating an array w/ null values rather than just creating it w/ the values once they're ready.